samples/cpp/calibration.cpp

```diff
@@ -529,6 +543,40 @@ int main( int argc, char** argv )
             case ASYMMETRIC_CIRCLES_GRID:
                 found = findCirclesGrid( view, boardSize, pointbuf, CALIB_CB_ASYMMETRIC_GRID );
                 break;
+            case CHARUCOBOARD:
+            {
+                cv::aruco::Dictionary dictionary;
+                if (dictFilename == "None") {
+                    dictionary = cv::aruco::getPredefinedDictionary(arucoDict);
+                }
+                else {
+                    cv::FileStorage dict_file(dictFilename, cv::FileStorage::Mode::FORMAT_JSON && cv::FileStorage::Mode::READ);
+                    cv::FileNode fn(dict_file.root());
+                    dictionary.readDictionary(fn);
+                }
+                cv::Ptr <cv::aruco::CharucoBoard> board =
+                    cv::makePtr<cv::aruco::CharucoBoard>(cv::aruco::CharucoBoard({ boardSize.width + 1,boardSize.height + 1 },
+                                                squareSize, markerSize, dictionary));
+                std::vector<int> markerIds;
+                std::vector<std::vector<cv::Point2f>> markerCorners;
+                cv::aruco::ArucoDetector detector(dictionary);
+                detector.detectMarkers(view, markerCorners, markerIds);
+
+                // if at least one marker detected
+                if (markerIds.size() > 0) {
+                    std::vector<cv::Point2f> charucoCorners;
+                    std::vector<int> charucoIds;
+                    found = cv::aruco::interpolateCornersCharuco(markerCorners, markerIds, view, board, pointbuf, charucoIds);
+                    
+                    if (pointbuf.size() < boardSize.height*boardSize.width) {
+                        found = false;
+                    }
+                }
```

Comment:
No need to create yet another detector. CharucoBoard has `detectBoard` method. It does all you need.